vkcs/internal/services/networking/v2/routers/requests.go

```diff
@@ -18,9 +32,69 @@ func (opts RouterCreateOpts) ToRouterCreateMap() (map[string]interface{}, error)
 	return util.BuildRequest(opts, "router")
 }
 
-func Create(c *gophercloud.ServiceClient, opts routers.CreateOptsBuilder) routers.CreateResult {
-	r := routers.Create(c, opts)
-	r.Err = util.ErrorWithRequestID(r.Err, r.Header.Get(util.RequestIDHeader))
+type ExpectedNeutronError struct {
+	ErrCode int
+	ErrType string
+}
+
+func retryNeutronError(actual error, retryableErrors []ExpectedNeutronError) bool {
+	for _, expectedErr := range retryableErrors {
+		gophercloudErr, ok := errutil.As(actual, expectedErr.ErrCode)
+		if !ok {
+			continue
+		}
+		if expectedErr.ErrType == "" {
+			return true
+		}
+
+		neutronError, decodeErr := inetworking.DecodeNeutronError(gophercloudErr.Body)
+		if decodeErr != nil {
+			continue
+		}
+		if expectedErr.ErrType == neutronError.Type {
+			return true
+		}
+	}
+
+	return false
+}
+
+func CreateWithRetry(c *gophercloud.ServiceClient, opts routers.CreateOptsBuilder) routers.CreateResult {
```

Comment:
Not quite what I proposed, I supposed that you will get rid of this function and place retrying logic inside func `Create` for router resource.